src/corelibs/U2Gui/src/util/project/ProjectTreeController.cpp

```diff
@@ -792,8 +795,64 @@ void ProjectTreeController::sl_onRemoveSelectedItems() {
     const QList<Document *> selectedDocs = getDocsInSelection(deriveDocsFromObjs).values();
     const QList<Folder> selectedFolders = getSelectedFolders();
     const QList<GObject *> selectedObjects = objectSelection.getSelectedObjects();
+    
+    QStringList selectedItemNames;
+    foreach(Document * doc, selectedDocs) {
+        selectedItemNames.append(doc->getName());
+    }
+    foreach(const Folder& folder, selectedFolders) {
+        selectedItemNames.append(folder.getFolderName());
+    }
+    foreach(GObject* obj, selectedObjects) {
+        selectedItemNames.append(obj->getGObjectName());
+    }
+    
+    QString fullItemsNamesList;
+
+    QString warningMessageText = (1 == selectedItemNames.count()) ? tr("Remove selected item?") : tr("Remove selected items?");
+    warningMessageText += "<ul style=\"margin-top:5px;margin-bottom:0px\"><li>";
+
+    const bool tooManyItemsSelected = (MAX_DISPLAING_NAME_COUNT < selectedItemNames.count());
```

Comment:
MAX_DISPLAING_NAME_COUNT is a constant and can't be < or >.
To make it more readable (in English) I would suggest avoiding the use of Yoda style if not necessary:
The more readable text will be:  'selectedItemNames.count() < MAX_DISPLAING_NAME_COUNT'

Reply:
Why don't we want to allow removal of items > MAX_DISPLAING_NAME_COUNT ?